frontend/packages/console-dynamic-plugin-sdk/src/extensions/storage-class-provisioner.ts

```diff
@@ -0,0 +1,60 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { CodeRef, EncodedCodeRef, UpdateExtensionProperties } from '../types';
+
+export type ProvisionerProps = {
+  onParamChange: (id: string) => void;
+};
+
+export namespace ExtensionProperties {
+  export type StorageClassProvisioner = {
+    [key: string]: {
+      [key: string]: {
+        title: string;
+        provisioner: string;
+        allowVolumeExpansion: boolean;
+        parameters: {
+          [key: string]: {
+            name: string;
+            hintText: string;
+            value?: string;
+            visible?: boolean | EncodedCodeRef;
```

Comment:
Why `| EncodedCodeRef`. Looks like this is only used as a boolean.

Reply:
We are not using it as a function now. But it can be passed as a function, the function would take in storage class related items as params.